src/pipdeptree/_models/package.py

```diff
@@ -118,13 +123,18 @@ class DistPackage(Package):
 
     """
 
-    def __init__(self, obj: DistInfoDistribution, req: ReqPackage | None = None) -> None:
+    def __init__(self, obj: Distribution, req: ReqPackage | None = None) -> None:
         super().__init__(obj)
         self.req = req
         self._project_name = ""
 
     def requires(self) -> list[Requirement]:
-        return self._obj.requires()  # type: ignore[no-untyped-call,no-any-return]
+        req_list = []
+        if self._obj.requires:
+            for r in self._obj.requires:
+                req_list.append(Requirement(r))
+            return req_list
+        return None
```

Comment:
We could use a list comprehension here:
```suggestion
        return [Requirement(r) for r in self._obj.requires] if self._obj.requires else []
```

Reply:
fixed.

Reply:
Continuing this [discussion](https://github.com/tox-dev/pipdeptree/pull/333#discussion_r1527878895) here.

If I'm correct in that `Distribution.requires()` is giving us `Requirement` objects where some may represent the same package, I think one solution is to keep track of the requirement package names we have added and only append to the requirement list if it's we have not seen it before.

Reply:
You are right. The solution you mentioned I have considered. But finally I choose to keep the extras because I think the extra information seems somewhat useful, as it lets us know under what circumstances these dependencies will be installed.

Reply:
I believe this is optional. I can add a deduplication feature if you think it's better to only keep one.

Reply:
I would say there cannot be duplicates (it would also duplicate when generating frozen requirements).

We would need to handle the case where we are given requirement strings "foo >= 2.0" and "foo; extra=bar". Which  string should we choose to display? I would say we need the first needs to be prioritized (meaning if we have a dependency that's not optional, choose that over an extra).

... as I am writing this comment, I realized we don't need to consider extras-related requirements at all; just look for `Requirement` objects that aren't associated to any extras. We should be fetching them using `distributions()` so they will still be displayed, and we can figure out how to support extras when get around to it.

Reply:
Ok, I remove the extra, and add filter in `from_pkgs` for skip the req which already in `reqs`.